### PR TITLE
Update cheque cell

### PR DESCRIPTION
**Pull request by duanyytop**

### CODE REVIEW

File: contracts/ckb-cheque-script/src/entry/helper.rs [hunk 1]
```diff
@@ -51,3 +51,23 @@ pub fn check_witness_args(position: usize) -> Result<(), Error>{
...
+                                .code_hash(CODE_HASH_SECP256K1_BLAKE160.pack())
+                                .args(public_key_hash.pack())
+                                .hash_type(Byte::new(TYPE))
+                                .build();
+
+  if lock_hash != &hash::blake2b_160(lock_script.as_slice()) {
+      return Err(Error::WrongPubKey);
+  }
+  Ok(())
+}
```
Comment (shaojunda) [inline contracts/ckb-cheque-script/src/entry/helper.rs]: A trailing blank line is missing here.

File: tests/src/entry_tests.rs [hunk 2]
```diff
@@ -175,3 +368,24 @@ Some(Bytes::from(
...
+    );
+    let tx = context.complete_tx(tx);
+
+    let err = context.verify_tx(&tx, MAX_CYCLES).unwrap_err();
+    let script_cell_index = 0;
+    assert_error_eq!(
+        err,
+        ScriptError::ValidationFailure(NO_MATCHED_SIGNATURE).input_lock_script(script_cell_index)
+    );
+}
```
Comment (shaojunda) [inline tests/src/entry_tests.rs]: A trailing blank line is missing here.

File: tests/src/withdraw_tests.rs [hunk 3]
```diff
@@ -229,3 +398,45 @@ Bytes::from(
...
+    let tx = context.complete_tx(tx);
+
+    // run
+    let err = context.verify_tx(&tx, MAX_CYCLES).unwrap_err();
+    let script_cell_index = 0;
+    assert_error_eq!(
+        err,
+        ScriptError::ValidationFailure(WITHDRAW_CHEQUE_INPUT_SINCE_ERROR).input_lock_script(script_cell_index)
+    );
+}
```
Comment (shaojunda) [inline tests/src/withdraw_tests.rs]: A trailing blank line is missing here

File: contracts/ckb-cheque-script/src/entry.rs [hunk 4]
```diff
@@ -23,24 +21,34 @@ pub fn main() -> Result<(), Error> {
...
+        } else {
+            Err(Error::NoMatchedInputs)
+        }
     } else {
-        Err(Error::NoMatchedInputs)
+        // Validate the signatures of receiver and sender
+        let mut context = unsafe{ CKBDLContext::<[u8; 128 * 1024]>::new()};
+        let lib = LibSecp256k1::load(&mut context);
+        match helper::validate_blake2b_sighash_all(&lib, &receiver_lock_hash) {
+            Ok(_) => claim::validate(&sender_lock_hash, &receiver_lock_hash, false),
```
Comment (shaojunda) [inline contracts/ckb-cheque-script/src/entry.rs:44]: This means that if it is not a claim, it is withdraw, right?

Comment (duanyytop) [reply contracts/ckb-cheque-script/src/entry.rs:44]: Yes

Comment (shaojunda) [reply contracts/ckb-cheque-script/src/entry.rs:44]: OK

**Review by shaojunda — APPROVED** [review]

**Pull request merged**